java/arcs/sdk/testing/BaseTestHarness.kt

```diff
@@ -0,0 +1,92 @@
+package arcs.sdk.testing
+
+import arcs.core.data.HandleMode
+import arcs.core.host.EntityHandleManager
+import arcs.core.storage.driver.RamDisk
+import arcs.core.storage.driver.RamDiskDriverProvider
+import arcs.core.storage.handle.HandleManager
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.jvm.util.JvmTime
+import arcs.sdk.EntitySpec
+import arcs.sdk.Handle
+import arcs.sdk.Particle
+import arcs.sdk.testing.HandleFlavor.COLLECTION
+import arcs.sdk.testing.HandleFlavor.SINGLETON
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Indicates the flavor of a handle: Singleton or a Collection (more may come).
+ */
+enum class HandleFlavor {
+    SINGLETON,
+    COLLECTION
+}
+
+/**
+ * Groups all the per-handle information that's relevant to the TestHarness.
```

Comment:
As a user of the testing sdk, and a first-time reader of this - I don't know what is supposed to be relevant to the test harness. I think some more detail would be good in the KDoc here.

Reply:
I've moved these 2 types to be inner types of the BaseTestHarness, these are for internal wiring, not for the user of this class.

I added a bit more kdoc to the main class.